aleo.abnf

```diff
@@ -354,9 +338,13 @@ is = is-op ws operand ws operand ws %s"into" ws register
 
 assert = assert-op ws operand ws operand
 
-commit = commit-op ws operand ws operand ws %s"into" ws register
+commit = commit-op ws operand ws operand
+         ws %s"into" ws register
+         ws %s"as" ws ( %s"address" / %s"field" / %s"group" )
 
-hash = hash-op ws operand ws %s"into" ws register
+hash = hash-op ws operand
+       ws %s"into" ws register
+       ws %s"as" ws ( arithmetic-type / %s"address" )
```

Comment:
```suggestion
       ws %s"as" ws ( arithmetic-type / address-type )
```

Reply:
@bendyarm Excellent idea. I've done that also for `field-type` and `group-type`. As explained in the commit message, while equivalent, it makes the role slightly more clear. And the grammar is slightly better factored this way.